builtin/logical/pki/path_acme_test.go

```diff
@@ -243,8 +243,21 @@ func TestAcmeBasicWorkflow(t *testing.T) {
 			csrKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 			require.NoError(t, err, "failed generated key for CSR")
 
-			// Validate we reject CSRs that contain names that aren't in the original order
+			// Validate we reject CSRs that contain CN that aren't in the original order
 			badCr := &x509.CertificateRequest{
+				Subject:  pkix.Name{CommonName: "not-in-original-order.com"},
+				DNSNames: []string{identifiers[0], identifiers[1]},
+			}
+			t.Logf("csr: %v", badCr)
+
+			csrWithBadCName, err := x509.CreateCertificateRequest(rand.Reader, badCr, csrKey)
+			require.NoError(t, err, "failed generating csr with bad common name")
+
+			_, _, err = acmeClient.CreateOrderCert(testCtx, createOrder.FinalizeURL, csrWithBadCName, true)
+			require.Error(t, err, "should not be allowed to csr with different common names than order")
+
+			// Validate we reject CSRs that contain DNS names that aren't in the original order
```

Comment:
I think there's one more, `IPAddresses` if you want to be thorough, which would be a good idea I think :-)

Reply:
added